### DISCUSSION TITLE
Renovate doesn't rebase on new default branch and doesn't update target branch in Merge Request

### DISCUSSION THREAD

**Discussion by chkpnt** [renovatebot/renovate#15855] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.71.0

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

14.8.2

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

When I change the default branch in GitLab, the still open MRs are rebased on the old branch and their target isn't updated.

Use case:
Our branching model doesn't use a "main", "develop" or "next" branch, but release-branches only, where the release branch for the next release is the default branch. When the release-branch is changed, e.g. from `release/1.0.x` to `release/1.1.x`, the target of Renovate's MRs, which were opened when `release/1.0.x` was the default branch, keep their target branch despite being rebased on `release/1.1.x`.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
https://github.com/renovatebot/renovate/blob/main/docs/development/behavior.md#avoid-unjustified-bug-reports

**Reply by chkpnt** [reply]
Sorry, should have read this first. "I didn't expect this to happen" indeed applies to my finding.

**Comment by chkpnt** [comment]
I'm still having this issue. After changing the default branch, the open MRs are listed as "blocked" in the dependency dashboard. When I check the checkmark to start all over again, the branch are rebased on the new default branch, but the target branch is still the old branch. It seems I have to close all open MRs and force renovate to recreate the MRs.

**Reply by rarkins** [reply]
Renovate doesn't currently have functionality to rebase PRs when the default branch changes. You are welcome to submit a feature request issue, including steps to reproduce (I don't think a reproduction repo will help because PRs are not preserved once we fork it).

As a workaround today, you can rename (e.g. add `-old` to the PR title) and then close existing PRs.

**Comment by Xerkus** [comment]
We encountered this too.

We use automerge for lockfile maintenance and in our branching model branches are named after minor versions eg `1.2.x`. Default branch is always pointing to the next minor.

Since CI failed for lockfile maintenance renovate opened PR from then default `3.21.x`. Afterwards we had 4.0.0-rc release and default branch changed to `4.0.x`. Meanwhile renovate kept updating PR from the current default branch and since 4.0.x has CI issues resolved, the PR got green status. Renovate then proceeded to automerge PR. All is fine here, except for a small problem that PR base branch was still `3.21.x` but content from new default 4.0.x.
In effect we got BC breaking changes backported to previous major in fully automated manner.

That was 3 weeks ago and we just got BC break report from a user that caught us unaware. 

I believe this should be treated as a bug. PR should be updated from PR base branch or base branch should be changed or PR closed and recreated against branch renovate is using.

**Reply by internalsystemerror** [reply]
Why the thumbs down? Is renovate only designed with a non changing default branch in mind as I tried looking for confirmation of this and was unable to find anything. Renovate allows for multiple base branches so I would have thought carrying commits from a different branch into the one the PR is for, would be a bug?

**Reply by rarkins** [reply]
Thumbs down for "I believe this should be treated as a bug."

See https://github.com/renovatebot/renovate/blob/main/docs/development/help-us-help-you.md#avoid-unjustified-bug-reports

**Reply by rarkins** [reply]
Being able to change the default branch and have PRs automatically rebase is a feature request, found in #19900

**Reply by internalsystemerror** [reply]
Agreed, but in my mind as a lay person, getting commits from a different branch than the one being merged to is a bug. But thank you for clarifying, I understand now that changing the default branch is not something that renovate really supports right now, so I'll coordinate anything further on the PR linked above 👍

**Reply by Xerkus** [reply]
@rarkins The bug I see is in renovate moving changes from one branch to another. Interplay with automerge is potentially devastating.
> In effect we got BC breaking changes backported to previous major in fully automated manner.

It would be satisfactory if renovate were to refuse to update PR entirely with message similar to edited/blocked notification when non-renovate changes are detected. I don't think it is on automerge here to block the PR.

**Reply by viceice** [reply]
it's not a bug. if you do branch automerge, the error wouldn't occure.

you issue happens because renovate currently can't change the PR target when default branch changes. that's a feature which is not yet supported and there is an open issue for that feature request.

**Reply by Xerkus** [reply]
Ok, I see. Thank you for explaining your stance on this.

Branch automerge indeed works flawlessly. As far as I can tell there is no way to enable automerge for branch but disable when fallback PR was created due to CI failure?

**Comment by chkpnt** [comment, marked as answer]
This feature is available since v35.155.0, see https://github.com/renovatebot/renovate/pull/23010 which implements #19900.
